codec-http/src/main/java/io/netty/handler/codec/http/HttpServerUpgradeHandler.java

```diff
@@ -296,7 +296,7 @@ private boolean upgrade(final ChannelHandlerContext ctx, final FullHttpRequest r
 
         // Ensure that all required protocol-specific headers are found in the request.
         for (String requiredHeader : requiredHeaders) {
-            if (!request.headers().contains(requiredHeader)) {
+            if (!request.headers().contains((CharSequence) requiredHeader)) {
```

Comment:
There is a `contains(String name)` that I left deprecated in favor of the `contains(CharSequence name)`. As the code is now this cast avoids a deprecation warning.